### PR TITLE
Initial Extraction

### PR DESCRIPTION
**Pull request by dignifiedquire**
@whyrusleeping I can't reopen the other PR, no idea how it got closed. This should have the right history now.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package fslock
```
Comment (whyrusleeping) [inline main.go]: I wouldnt call it main.go. Maybe fslock.go? main is semantically for binaries.

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if isLocked != expected {
+		t.Fatalf("expected %t to be %t", isLocked, expected)
+	}
+}
+
+func TestLock_Simple(t *testing.T) {
```
Comment (whyrusleeping) [inline main_test.go]: underbars...............

Comment (dignifiedquire) [reply main_test.go]: _______________ just for you

Comment (dignifiedquire) [reply main_test.go]: done

File: main.go (hunk 1 above)
Comment (dignifiedquire) [reply main.go]: done

**Comment by dignifiedquire** [comment]
@magik6k done

**Pull request merged**